src/org/connection.ts

```diff
@@ -82,6 +82,12 @@ export class Connection<S extends Schema = Schema> extends JSForceConnection<S>
   // All connections are tied to a username
   private username!: string;
 
+  // Save the DNS resolution result of this connection's instance URL.
+  private resolvable!: boolean;
+
+  // Save the max API version of this connection's org.
+  private maxApiVersion!: string;
+
```

Comment:
let's avoid the untrue TS assertions.  It's better to check for the definedness of these props each time than to guarantee they're always defined.

you could always set `resolvable` to false, then make it true you've resolved it once.  (maybe a better name is hasResolved?)

maxApiVersion could definitely be undefined.